src/Microsoft.OData.Core/Json/JsonValueUtilsAsync.cs

```diff
@@ -0,0 +1,491 @@
+                        //  [("+" / "-") offset]
+                        //  quotation-mark
+                        //
+                        // offset = 4DIGIT
+                        string textValue = XmlConvert.ToString(value);
+                        await writer.WriteQuotedAsync(textValue).ConfigureAwait(false);
+                    }
+
+                    break;
+
+                case ODataJsonDateTimeFormat.ODataDateTime:
+                    {
+                        // Uses the same format as DateTime but with offset:
+                        // jsonDateTime= quotation-mark
+                        //  "\/Date("
+                        //  ticks
+                        //  [("+" / "-") offset]
+                        //  ")\/"
+                        //  quotation-mark
+                        //
+                        // ticks = *DIGIT
+                        // offset = 4DIGIT
+                        string textValue = String.Format(
+                            CultureInfo.InvariantCulture,
+                            JsonConstants.ODataDateTimeOffsetFormat,
+                            DateTimeTicksToJsonTicks(value.Ticks),
+                            offsetMinutes >= 0 ? JsonConstants.ODataDateTimeOffsetPlusSign : string.Empty,
+                            offsetMinutes);
+                        await writer.WriteQuotedAsync(textValue).ConfigureAwait(false);
+                    }
+
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Write a TimeSpan value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">TimeSpan value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, TimeSpan value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            await writer.WriteQuotedAsync(EdmValueWriter.DurationAsXml(value)).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Write a TimeOfDay value
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">TimeOfDay value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, TimeOfDay value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            await writer.WriteQuotedAsync(value.ToString()).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Write a Date value
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">Date value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, Date value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            await writer.WriteQuotedAsync(value.ToString()).ConfigureAwait(false);

```

Comment:
Add CultureInfo

Reply:
@KenitoInc `.ToString()` against a `byte` suffices here

Reply:
@KenitoInc The `value` here is of type `Microsoft.OData.Edm.Date` and the custom `.ToString()` method already takes culture into consideration [here](https://github.com/OData/odata.net/blob/8444960bed25590dcdb6398b86cd5f7e3acef7a5/src/Microsoft.OData.Edm/Schema/Date.cs#L239)